plugins/inputs/socketstat/socketstat_test.go

```diff
@@ -0,0 +1,123 @@
+		proto    []string
+		filename string
+		tags     []map[string]string
+		fields   [][]map[string]interface{}
+		err      error
+	}{
+		{
+			name:     "tcp - no sockets => no results",
+			proto:    []string{"tcp"},
+			filename: "tcp_no_sockets.txt",
+		},
+		{
+			name:     "udp - no sockets => no results",
+			proto:    []string{"udp"},
+			filename: "udp_no_sockets.txt",
+		},
+		{
+			name:     "tcp sockets captured",
+			proto:    []string{"tcp"},
+			filename: "tcp_traffic.txt",
+			tags: []map[string]string{
+				{"proto": "tcp", "local_addr": "192.168.1.21", "local_port": "6514", "remote_addr": "192.168.1.21", "remote_port": "443"},
+				{"proto": "tcp", "local_addr": "192.168.122.1", "local_port": "55194", "remote_addr": "192.168.122.1", "remote_port": "6514"},
+				{"proto": "tcp", "local_addr": "127.0.0.1", "local_port": "7778", "remote_addr": "127.0.0.1", "remote_port": "50378"},
+			},
+			fields: [][]map[string]interface{}{
+				{map[string]interface{}{"state": "ESTAB", "bytes_acked": uint64(1126), "bytes_received": uint64(532644751), "segs_out": uint64(211249), "segs_in": uint64(211254), "data_segs_out": uint64(2), "data_segs_in": uint64(211251), "recv_q": uint64(0), "send_q": uint64(0)}},
+				{map[string]interface{}{"state": "ESTAB", "bytes_acked": uint64(790782896), "bytes_received": uint64(1126), "segs_out": uint64(333361), "segs_in": uint64(333361), "data_segs_out": uint64(333358), "data_segs_in": uint64(2), "recv_q": uint64(0), "send_q": uint64(0)}},
+				{map[string]interface{}{"state": "ESTAB", "bytes_acked": uint64(19983121), "bytes_received": uint64(266383), "segs_out": uint64(15431), "segs_in": uint64(17633), "data_segs_out": uint64(15119), "data_segs_in": uint64(5098), "recv_q": uint64(0), "send_q": uint64(0)}},
+			},
+		},
+		{
+			name:     "udp packets captured",
+			proto:    []string{"udp"},
+			filename: "udp_traffic.txt",
+			tags: []map[string]string{
+				{"proto": "udp", "local_addr": "10.10.0.4", "local_port": "33149", "remote_addr": "10.10.0.5", "remote_port": "53"},
+				{"proto": "udp", "local_addr": "10.10.0.4", "local_port": "54276", "remote_addr": "10.10.0.6", "remote_port": "53"},
+				{"proto": "udp", "local_addr": "10.10.0.4", "local_port": "38312", "remote_addr": "10.10.0.7", "remote_port": "53"},
+			},
+			fields: [][]map[string]interface{}{
+				{map[string]interface{}{"recv_q": uint64(0), "send_q": uint64(0)}},
+				{map[string]interface{}{"recv_q": uint64(0), "send_q": uint64(0)}},
+				{map[string]interface{}{"recv_q": uint64(0), "send_q": uint64(0)}},
+			},
+		},
+	}
+	for i, tt := range tests {
+		octets, err := os.ReadFile(filepath.Join("testdata", tt.filename))
+		require.NoError(t, err)
+
+		t.Run(tt.name, func(t *testing.T) {
+			i++
+			ss := &Socketstat{
+				SocketProto: tt.proto,
+			}
+			acc := new(testutil.Accumulator)
+
+			err := ss.Init()
+			if err != nil && !strings.Contains(err.Error(), "executable file not found in $PATH") {
+				require.NoError(t, err)
+			}
```

Comment:
I think this is easier to understand
```suggestion
			if err != nil {
				require.EqualError(t, err, "executable file not found in $PATH")
			}
```